cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py

```diff
@@ -0,0 +1,473 @@
+from cfme import test_requirements
+from cfme.base.credential import Credential
+from cfme.common.vm import VM
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.intelligence.reports.reports import CustomReport
+from cfme.markers.env_markers.provider import ONE
+from cfme.utils.log import logger
+from cfme.utils.wait import wait_for
+
+pytestmark = [
+    pytest.mark.tier(2),
+    pytest.mark.parametrize('interval', ['Daily', 'Weekly', 'Monthly'],
+        ids=['daily_rate', 'weekly_rate', 'monthly_rate'], scope='module'),
+    pytest.mark.provider([VMwareProvider], selector=ONE,
+                       scope='module',
+                       required_fields=[(['cap_and_util', 'test_chargeback'], True)]),
+    pytest.mark.usefixtures('has_no_providers_modscope', 'setup_provider_modscope'),
+    test_requirements.chargeback,
+]
+
+# Allowed deviation between the reported value in the Chargeback report and the estimated value.
+DEVIATION = 1
+
+
+@pytest.yield_fixture(scope="module")
+def vm_ownership(enable_candu, provider, appliance):
+    # In these tests, chargeback reports are filtered on VM owner.So,VMs have to be
+    # assigned ownership.
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    if not provider.mgmt.does_vm_exist(vm_name):
+        pytest.skip("Skipping test, cu-24x7 VM does not exist")
+    if not provider.mgmt.is_vm_running(vm_name):
+        provider.mgmt.start_vm(vm_name)
+        provider.mgmt.wait_vm_running(vm_name)
+
+    group_collection = appliance.collections.groups
+    cb_group = group_collection.instantiate(description='EvmGroup-user')
+
+    vm = VM.factory(vm_name, provider)
+    user = None
+    try:
+        user = appliance.collections.users.create(
+            name=provider.name + fauxfactory.gen_alphanumeric(),
+            credential=Credential(principal='uid' + '{}'.format(fauxfactory.gen_alphanumeric()),
+                secret='secret'),
+            email='abc@example.com',
+            groups=cb_group,
+            cost_center='Workload',
+            value_assign='Database')
+        vm.set_ownership(user=user.name)
+        logger.info('Assigned VM OWNERSHIP for {} running on {}'.format(vm_name, provider.name))
+
+        yield user.name
+    finally:
+        vm.unset_ownership()
+        if user:
+            user.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def enable_candu(appliance):
+    # C&U data collection consumes a lot of memory and CPU.So, we are disabling some server roles
+    # that are not needed for Chargeback reporting.
+
+    candu = appliance.collections.candus
+    server_info = appliance.server.settings
+    original_roles = server_info.server_roles_db
+    server_info.enable_server_roles(
+        'ems_metrics_coordinator', 'ems_metrics_collector', 'ems_metrics_processor')
+    server_info.disable_server_roles('automate', 'smartstate')
+    candu.enable_all()
+
+    yield
+
```

Comment:
**REQUIRED**
Here and further please remove these blank lines around `yield`.

Reply:
@quarckster I would disagree with this requirement, as it isn't in our style guide.  Further, I'd disagree with trying to enforce this, as I believe it falls within PEP8's guidelines for use of blank lines.  Personally I like this style for yield fixtures, or yielding methods in general, as it cleanly separates the logical blocks of the function, and makes it painfully clear that it yelds, not returns.

https://www.python.org/dev/peps/pep-0008/#blank-lines

Reply:
I too agree with @mshriver.   I was taught early on to group code kind of like we group paragraphs in human languages, using the blank lines to separate the big things we are doing.

Reply:
I won't insist, but I don't like it :-P

Reply:
let me add two words from my side :). I'd ask to remove first empty line between last statement of first code block and yield since this is one logical block. however, empty line after yield is really necessary because it splits two blocks of logic and makes code more readable.